src/AdminConsole/Services/MagicLinks/MagicLinkSignInManager.cs

```diff
@@ -1,73 +1,49 @@
 using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Identity;
-using Microsoft.AspNetCore.Mvc;
-using Microsoft.AspNetCore.Mvc.Infrastructure;
-using Microsoft.AspNetCore.Mvc.Routing;
 using Microsoft.Extensions.Options;
 using Passwordless.AdminConsole.EventLog.Loggers;
 using Passwordless.AdminConsole.Identity;
 using Passwordless.AdminConsole.Services.Mail;
 using SignInResult = Microsoft.AspNetCore.Identity.SignInResult;
 
-namespace Passwordless.AdminConsole.Services;
+namespace Passwordless.AdminConsole.Services.MagicLinks;
 
 public class MagicLinkSignInManager<TUser> : SignInManager<TUser> where TUser : class
 {
-    public const string PasswordlessSignInPurpose = "MagicLinkSignIn";
-
+    private readonly IMagicLinkBuilder _magicLinkBuilder;
     private readonly IMailService _mailService;
-    private readonly IUrlHelperFactory _urlHelperFactory;
-    private readonly IActionContextAccessor _actionContextAccessor;
     private readonly IEventLogger _eventLogger;
 
-    public MagicLinkSignInManager(UserManager<TUser> userManager,
+    public MagicLinkSignInManager(
+        IMagicLinkBuilder magicLinkBuilder,
+        UserManager<TUser> userManager,
         IHttpContextAccessor contextAccessor,
         IUserClaimsPrincipalFactory<TUser> claimsFactory,
         IOptions<IdentityOptions> optionsAccessor,
         ILogger<SignInManager<TUser>> logger,
         IAuthenticationSchemeProvider schemes,
         IUserConfirmation<TUser> confirmation,
-        IMailService mailService, IUrlHelperFactory urlHelperFactory, IActionContextAccessor actionContextAccessor,
+        IMailService mailService,
         IEventLogger eventLogger)
         : base(userManager, contextAccessor, claimsFactory, optionsAccessor, logger, schemes, confirmation)
     {
+        _magicLinkBuilder = magicLinkBuilder;
         _mailService = mailService;
-        _urlHelperFactory = urlHelperFactory;
-        _actionContextAccessor = actionContextAccessor;
         _eventLogger = eventLogger;
     }
 
     public async Task<SignInResult> SendEmailForSignInAsync(string email, string? returnUrl)
```

Comment:
This result is never used, I think we can remove it since it doesn't add any value and is always successful.

Reply:
I'll add an item to investigate this